modules/core/import/modules/csv/src/Access/CsvImportMigrationAccess.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Drupal\farm_import_csv\Access;
+
+use Drupal\Component\Plugin\Exception\PluginNotFoundException;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\migrate\Plugin\MigrationPluginManager;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Access checking logic for CSV importers.
+ */
+class CsvImportMigrationAccess {
+
+  /**
+   * The migration plugin manager.
+   *
+   * @var \Drupal\migrate\Plugin\MigrationPluginManager
+   */
+  protected $pluginManagerMigration;
+
+  /**
+   * MigrateSourceUiForm constructor.
+   *
+   * @param \Drupal\migrate\Plugin\MigrationPluginManager $plugin_manager_migration
+   *   The migration plugin manager.
+   */
+  public function __construct(MigrationPluginManager $plugin_manager_migration) {
+    $this->pluginManagerMigration = $plugin_manager_migration;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('plugin.manager.migration'),
+    );
+  }
```

Comment:
I do not believe this `create()` method is necessary since we are not implementing the ContainerFactory interface, instead the dependencies are wired in `services.yml`

Reply:
https://github.com/farmOS/farmOS/pull/722/commits/ee8bfa597d055cb45bedaf822c7c65525b9f30b6